KLSadd_insertion/src/updateChapters.py

```diff
@@ -1,110 +1,327 @@
+    :return: The big word without the word that was searched for
+    """
+    a = word_to_find
+    b = word_to_search_in
+    precheck = 1
+    if a in b:
+        if "\\paragraph{\\bf KLS Addendum: " in b or "\\subsubsection*{\\bf KLS Addendum: " in b:
+            while True:
+                if "\\paragraph{\\bf KLS Addendum: " in b[b.find(a)-precheck:b.find(a)] or \
+                        "\\subsubsection*{\\bf KLS Addendum: " in b[b.find(a) - precheck:b.find(a)]:
+                    return b[:b.find(a) - precheck] + b[b.find(a) + len(a):]
+                else:
+                    precheck += 1
+
+        else:
+            cut = b[:b.find(a)] + b[b.find(a) + len(a):]
+            return cut
+    else:
+        return b
+
+
+def prepare_for_pdf(chap):
+    """
+    Edits the chapter string sent to include hyperref, xparse, and cite packages
+
+    :param chap: The chapter (9 or 14) that is being processed
+    :return: The processed chapter, ready for additional processing
+    """
+    foot_misc_index = 0
     index = 0
     for word in chap:
         index += 1
-        if("footmisc" in word):
-            footmiscIndex += index
-    # edits the chapter string sent to include hyperref, xparse, and cite packages
-    #str[footmiscIndex] += "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n"
-    chap.insert(
-        footmiscIndex,
-        "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n")
+        if "footmisc" in word:
+            foot_misc_index += index
+    # str[footmiscIndex] += "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n"
+    chap.insert(foot_misc_index, "\\usepackage[pdftex]{hyperref} \n\\usepackage {xparse} \n\\usepackage{cite} \n")
     return chap
 
-# 2/18/16 this method reads in relevant commands that are in KLSadd.tex
-# and returns them as a list
 
+def get_commands(kls, new_commands):
+    """
+    This method reads in relevant commands that are in KLS Addendum.tex and returns them as a list
 
-def getCommands(kls):
+    :param kls: The addendum is the source of what is being found
+    :return:
+    """
     index = 0
     for word in kls:
         index += 1
-        if("smallskipamount" in word):
-            newCommands.append(index - 1)
-        if("mybibitem[1]" in word):
-            newCommands.append(index)
-    comms = kls[newCommands[0]:newCommands[1]]
-    return comms
+        if "smallskipamount" in word:
+            new_commands.append(index - 1)
+        if "mybibitem[1]" in word:
+            new_commands.append(index)
+    print new_commands
+    return kls[new_commands[0]:new_commands[1]]
 
-# 2/18/16 this method addresses the goal of hardcoding in the necessary
-# commands to let the chapter files run as pdf's. Currently only works
-# with chapter 9
 
+# 2/18/16 this method addresses the goal of hardcoding in the necessary commands to let the chapter files run as pdf's.
```

Comment:
The dates of doing things don't belong in comments - this reads more like a commit message, which already has a date built into it. Also, whatever is in here should be just in the docstring, which does not say this stuff.